xcb-types.el

```diff
@@ -731,9 +729,8 @@ and the second the consumed length."
                                                (eieio-oref-default obj name)
                                                nil total-length))
                (setf (slot-value obj name) (car tmp))
-               (setq count (+ count (cadr tmp)))
-               (setq data (substring data (cadr tmp)))))))
-       (list initform count)))
+               (setq offset (+ offset (cadr tmp)))))))
+       (list initform offset)))
```

Comment:
Thanks, @Stebalien. The patch doesn't seem completely correct. Here the offset is returned and not a count. The other branches of unmarshal-field return a count.

Reply:
No, you're right. We need to return the count of bytes read, not the entire offset.

Although, arguably, it should return the offset everywhere.

Reply:
> Although, arguably, it should return the offset everywhere.

The docstring says that a count of consumed bytes (?) is returned. We could change this to an offset everywhere. All that matters is that we do it consistently.

Reply:
Fixed. Although I want to clean this all up in the future (there's no reason we can't use offsets for structs as well).

Reply:
I think we should clean this up before merging. Offset in, offset out, everywhere. This should simplify things slightly.

Reply:
Fair enough.

Reply:
I looked again into this. It is better if we go with the more minimal change proposed here.